dev/set_matrix.py

```diff
@@ -99,7 +99,7 @@ def get_released_versions(package_name):
         #
         # > pip install 'xgboost==0.7'
         # ERROR: Could not find a version that satisfies the requirement xgboost==0.7
-        if len(dist_files) > 0
+        if len(dist_files) > 0 and (not dist_files[0]["yanked"])
```

Comment:
All distribution files in a yanked releaese should be marked as `"yanked": true`.